src/services/goals.js

```diff
@@ -83,9 +83,18 @@ async function removeUnusedObjectivesGoalsFromReport(reportId, currentGoals, tra
   const previousObjectiveIds = previousObjectives.map((ro) => ro.objectiveId);
 
   const goalIdsToRemove = previousGoalIds.filter((id) => !currentGoalIds.includes(id));
-  const objectiveIdsToRemove = previousObjectiveIds
-    .filter((id) => !currentObjectiveIds.includes(id));
-
+  const objectiveIdsToRemove = [];
+  Promise.all(
+    previousObjectiveIds.map(async (id) => {
+      const notCurrent = !currentObjectiveIds.includes(id);
+      const activityReportObjectives = await ActivityReportObjective
+        .findAll({ where: { objectiveId: id } });
+      const lastInstance = activityReportObjectives.length <= 1;
+      if (notCurrent && lastInstance) {
+        objectiveIdsToRemove.push(id);
+      }
+    }),
+  );
   await removeActivityReportObjectivesFromReport(reportId, transaction);
```

Comment:
This is what was actually causing the 500's when people tried to remove the objectives that were added from previous reports. This filters them and only deletes objectives that are not in use by other reports.